components/eam/src/chemistry/yaml/mo_drydep/f90_yaml/calculate_resistance_rgsx_and_rsmx_beg_yml.f90

```diff
@@ -96,12 +98,29 @@
         call write_var(unit_input,unit_output,'tc',tc(yaml%col_print))
         call write_var(unit_input,unit_output,'heff',heff(yaml%col_print,:))
         call write_var(unit_input,unit_output,'crs',crs(yaml%col_print))
-        call write_var(unit_input,unit_output,'ri',ri)
-        call write_var(unit_input,unit_output,'rgso',rgso)
-        call write_var(unit_input,unit_output,'rgss',rgss)
-        call write_var(unit_input,unit_output,'foxd',foxd)
-        call write_var(unit_input,unit_output,'drat',drat)
-
+        call write_var(unit_input,unit_output,'ri',ri(:,:))
+        call write_var(unit_input,unit_output,'rgso',rgso(:,:))
+        call write_var(unit_input,unit_output,'rgss',rgss(:,:))
+        call write_var(unit_input,unit_output,'foxd',foxd(:))
+        call write_var(unit_input,unit_output,'drat',drat(:))
+
+        !  add additional input module fields here:
+
+        call write_var(unit_input,unit_output,'n_land_type',n_land_type)
+        call write_var(unit_input,unit_output,'gas_pcnst',gas_pcnst)
+        call write_var(unit_input,unit_output,'nddvels',nddvels)
+        call write_var(unit_input,unit_output,'so2_ndx',so2_ndx)
+        call write_var(unit_input,unit_output,'lt_for_water',lt_for_water)
+        call write_var(unit_input,unit_output,'large_value',large_value)
+        call write_var(unit_input,unit_output,'has_dvel',has_dvel(:))
+        call write_var(unit_input,unit_output,'map_dvel',map_dvel(:))
+
+        !  the following are intent(out) fields for this subroutine, so they are initialized to zero
+        !  here to facilitate testing when they are written as output.
+
+        cts(:) = 0._r8
+        rgsx(:,:,:) = 0._r8
+        rsmx(:,:,:) = 0._r8
 
```

Comment:
It's been a little while since I've written much of any Fortran, so wanted to check and make sure...

Since this file is `#include`d in the actual subroutine call, then these arrays will be initialized to zero in that enclosing subroutine, right? It looks that way to me

Reply:
Yes, this works. Ignore the Slack chat. I believe you can also do `rgsx = 0._r8` if you like but either syntax works.